src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueDebtAdaptor.sol

```diff
@@ -0,0 +1,244 @@
+     */
+    function borrowFromMorphoBlue(Id _id, uint256 _amountToBorrow) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        _borrowAsset(market, _amountToBorrow, address(this));
+
+        // Check if borrower is insolvent (AKA they have bad LTV), revert if they are
+        if (minimumHealthFactor > (_getHealthFactor(_id, market))) {
+            revert MorphoBlueDebtAdaptor__HealthFactorTooLow(_id);
+        }
+    }
+
+    // `repayDebt`
+
+    /**
+     * @notice Allows strategists to repay loan debt on Morph Blue Lending Market. Make sure to call addInterest() beforehand to ensure we are repaying what is required.
+     * @dev Uses `_maxAvailable` helper function, see BaseAdaptor.sol
+     * @param _id Encoded bytes32 MB id that represents the MB market for this position.
+     * @param _debtTokenRepayAmount The amount of `loanToken` to repay.
+     */
+    function repayMorphoBlueDebt(Id _id, uint256 _debtTokenRepayAmount) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        _accrueInterest(market); // either call this or use periphery library.
+
+        ERC20 tokenToRepay = ERC20(market.loanToken);
+        uint256 debtAmountToRepay = _maxAvailable(tokenToRepay, _debtTokenRepayAmount);
+
+        // using Morpho sharesLibrary we can calculate the sharesToRepay from the debtAmount
+        uint256 totalBorrowAssets = morphoBlue.market(_id).totalBorrowAssets;
+        uint256 totalBorrowShares = morphoBlue.market(_id).totalBorrowShares;
+        uint256 sharesToRepay = debtAmountToRepay.toSharesUp(totalBorrowAssets, totalBorrowShares); // get the total assets and total borrow shares of the market
+
+        // MorphoBlue reverts w/ underflow/overflow error if trying to repay with more than we have. That said, we will manage when strategists try to pass in type(uint256).max
+        uint256 sharesAccToMorphoBlue = morphoBlue.borrowShares(_id, address(this));
+        if (sharesAccToMorphoBlue < sharesToRepay) {
+            sharesToRepay = sharesAccToMorphoBlue;
+        }
+        
+        tokenToRepay.safeApprove(address(morphoBlue), type(uint256).max);
+        _repayAsset(market, sharesToRepay, address(this));
+        _revokeExternalApproval(tokenToRepay, address(morphoBlue));
+    }
+
+    /**
+     * @notice Allows a strategist to call `accrueInterest()` on a MB Market cellar is using.
+     * @dev A strategist might want to do this if a MB market has not been interacted with
+     *      in a while, and the strategist does not plan on interacting with it during a
+     *      rebalance.
+     * @dev Calling this can increase the share price during the rebalance,
+     *      so a strategist should consider moving some assets into reserves.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     */
+    function accrueInterest(Id _id) public {
```

Comment:
Can this logic be moved into a helper file?

Reply:
See my comment [here](https://github.com/PeggyJV/cellar-contracts/pull/154/files#r1443878990).